### PR TITLE
convert osm to format class

### PR DESCRIPTION
**Pull request by tsteven4**

### CODE REVIEW

File: osm.cc [hunk 1]
```diff
@@ -789,9 +736,9 @@ osm_waypt_disp(const Waypoint* waypoint)
 
   waypoints.insert(name, waypoint);
 
-  int* id = (int*) xmalloc(sizeof(*id));
+  auto* id = new int;
   *id = --node_id;
-  (const_cast<Waypoint*>(waypoint))->extra_data = id;
+  const_cast<Waypoint*>(waypoint)->extra_data = id;
```
Comment (tsteven4) [inline osm.cc:741]: Codacy Issue found: [Assigning an integer to a pointer is not portable.](https://app.codacy.com/manual/gpsbabel/gpsbabel/pullRequest?prid=5617520)

**Comment by tsteven4** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Clones added
============
- osm.h  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/gpsbabel/gpsbabel/pullRequest?prid=5617520&bid=18402623)

**Comment by tsteven4** [comment]
The following cppcheck issue is a false positive with cppcheck 1.90.  It is fixed in cppcheck 2.0.

```
osm.cc:741:47: portability: Assigning an integer to a pointer is not portable. [AssignmentIntegerToAddress]
  const_cast<Waypoint*>(waypoint)->extra_data = i
```

**Comment by robertlipe** [comment]
It's probably ill-tested in cppcheck because it's a weird thing to do in
"real" c++.  I wonder if someday our extra_data needs should be addressed
by subclassing Waypoint. (And if Geocache shouldn't be the same.)

That day isn't today. Looks good to me, Steven.

RJL

**Pull request merged**